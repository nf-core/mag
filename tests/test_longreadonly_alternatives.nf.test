nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "test"
    profile "test_longreadonly_alternatives"

    test("-profile longreadonly_alternatives") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_mag_software_mqc_versions.yml"),
                ).match() }
            )
        }
    }
}
