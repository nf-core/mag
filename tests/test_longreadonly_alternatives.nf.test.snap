{
    "-profile longreadonly_alternatives": {
        "content": [
            26,
            {
                "ADJUST_MAXBIN2_EXT": {
                    "coreutils": 9.5
                },
                "BIN_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "BUSCO_BUSCO": {
                    "busco": "6.0.0"
                },
                "BUSCO_UNTAR": {
                    "untar": 1.34
                },
                "CONCAT_BINQC_TSV": {
                    "csvtk": "0.31.0"
                },
                "CONVERT_DEPTHS": {
                    "bioawk": 20110810
                },
                "FLYE": {
                    "flye": "2.9.5-b1801"
                },
                "GUNZIP_BINS": {
                    "gunzip": 1.13
                },
                "GUNZIP_LONGREAD_ASSEMBLIES": {
                    "gunzip": 1.13
                },
                "MAG_DEPTHS": {
                    "python": "3.6.7",
                    "pandas": "1.1.5"
                },
                "MAG_DEPTHS_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "MAXBIN2": {
                    "maxbin2": "2.2.7"
                },
                "METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS_LONGREAD": {
                    "metabat2": 2.15
                },
                "METABAT2_METABAT2": {
                    "metabat2": 2.17
                },
                "MINIMAP2_ASSEMBLY_ALIGN": {
                    "minimap2": "2.29-r1283",
                    "samtools": 1.21
                },
                "MINIMAP2_ASSEMBLY_INDEX": {
                    "minimap2": "2.29-r1283"
                },
                "NANOLYSE": {
                    "nanolyse": "1.2.0"
                },
                "NANOPLOT_FILTERED": {
                    "nanoplot": "1.46.1"
                },
                "NANOPLOT_RAW": {
                    "nanoplot": "1.46.1"
                },
                "NANOQ": {
                    "nanoq": "0.10.0"
                },
                "PORECHOP_PORECHOP": {
                    "porechop": "0.2.4"
                },
                "SEQKIT_STATS": {
                    "seqkit": "2.9.0"
                },
                "SPLIT_FASTA": {
                    "python": "3.6.7",
                    "biopython": "1.7.4",
                    "pandas": "1.1.5"
                },
                "Workflow": {
                    "nf-core/mag": "v5.1.0dev"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-21T21:30:38.319824244"
    },
    "multiqc": {
        "content": [
            [
                "multiqc/multiqc_data/BETA-multiqc.parquet",
                "multiqc/multiqc_data/busco_plot_bacteria_odb10.yaml",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_busco.yaml",
                "multiqc/multiqc_data/multiqc_citations.yaml",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_general_stats.yaml",
                "multiqc/multiqc_data/multiqc_software_versions.yaml",
                "multiqc/multiqc_data/multiqc_sources.yaml",
                "multiqc/multiqc_data/porechop.yaml",
                "multiqc/multiqc_plots/pdf/busco_plot_bacteria_odb10-cnt.pdf",
                "multiqc/multiqc_plots/pdf/busco_plot_bacteria_odb10-pct.pdf",
                "multiqc/multiqc_plots/png/busco_plot_bacteria_odb10-cnt.png",
                "multiqc/multiqc_plots/png/busco_plot_bacteria_odb10-pct.png",
                "multiqc/multiqc_plots/svg/busco_plot_bacteria_odb10-cnt.svg",
                "multiqc/multiqc_plots/svg/busco_plot_bacteria_odb10-pct.svg",
                "multiqc/multiqc_report.html"
            ],
            [
                "busco_plot_bacteria_odb10.yaml:md5,61ea14a5ac60fa3b48ee832af75effff",
                "multiqc_busco.yaml:md5,0a1896fc18fdea30852125167730c8c5",
                "multiqc_citations.yaml:md5,cd0615b89829a041bc2a33cdb69b585b",
                "multiqc_general_stats.yaml:md5,f5594b277c894e24b538c4ef9e468e3d",
                "porechop.yaml:md5,828d96e2822bb4b4108ab706c00c4450"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T11:26:00.0376308"
    }
}